ReactAndroid/src/main/java/com/facebook/react/modules/settings/SettingsModule.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+package com.facebook.react.modules.settings;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+
+import com.facebook.react.bridge.Arguments;
+import com.facebook.react.bridge.ReactApplicationContext;
+import com.facebook.react.bridge.ReactContextBaseJavaModule;
+import com.facebook.react.bridge.ReactMethod;
+import com.facebook.react.bridge.ReadableMap;
+import com.facebook.react.bridge.ReadableMapKeySetIterator;
+import com.facebook.react.bridge.ReadableType;
+import com.facebook.react.bridge.WritableMap;
+import com.facebook.react.module.annotations.ReactModule;
+import com.facebook.react.modules.core.DeviceEventManagerModule;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@ReactModule(name = SettingsModule.NAME)
+public class SettingsModule extends ReactContextBaseJavaModule  implements SharedPreferences.OnSharedPreferenceChangeListener{
+
+  @SuppressWarnings("WeakerAccess")
+  static final String NAME = "SettingsManager";
+  static final String DEFAULT_FILE_NAME = "ReactNative";
+
+  static private String sFilename = "";
+  private Boolean ignoringUpdates = false;
+  private SharedPreferences mPreferences;
+
+  public SettingsModule(ReactApplicationContext context) {
+    super(context);
+  }
+
+  @Override
+  public String getName() {
+    return SettingsModule.NAME;
+  }
+
+  private SharedPreferences getPreferences() {
+    if (mPreferences == null) {
+      if (SettingsModule.sFilename.length() > 0) {
+        mPreferences = getReactApplicationContext().getSharedPreferences(SettingsModule.sFilename, Context.MODE_PRIVATE);
+      } else if (getCurrentActivity() != null) {
+        mPreferences = getCurrentActivity().getPreferences(Context.MODE_PRIVATE);
```

Comment:
The settings file created by the Settings Module is accessible to that Settings module and therefore indirectly accessible to any Activity or to any other Java class that imports the Settings module.

The only potential concern raised by calling `getPreferences()` here is for apps that have multiple Activities each instantiating a MainReactPackage. This is by no means a common occurrence, but if an app is engaging in this behavior - then having each Settings Module in each MainReactPackage use its own, private settings store seems like the most reasonable default behavior. 

Nevertheless, if separate settings files are not desired, that is, if the developer whats to ensure that every Settings Module instantiated anywhere in the the app will always use the same settings file - then that developer merely has to provide a file name to the Settings Module, and - as long as they take care that only one Settings Module is ever extant at one time - it will be the case that every Settings Module created anywhere in the app will use the same settings file which will have the provided name.